libopendavinci/src/odcore/base/module/ManagedClientModule.cpp

```diff
@@ -116,12 +121,106 @@ namespace odcore {
                 return m_startOfLastCycle;
             }
 
+            std::vector<double> ManagedClientModule::computeAdvancedStatistics() {
+#ifdef __linux__
+                bool success=false;
+                double cpu_stats[4];
+                
+                try
+                {
+                    // get system info
+                    
+                    // The number of processors currently online
+                    int numCPU = sysconf(_SC_NPROCESSORS_ONLN);
+                    if(numCPU == -1) OPENDAVINCI_CORE_THROW_EXCEPTION(IOException,"Could not get system info");
```

Comment:
It is better to enclose the exception statement in brackets:
if(numCPU == -1){
    ...
}